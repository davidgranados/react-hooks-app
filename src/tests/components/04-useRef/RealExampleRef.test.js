import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe("Tests on RealExampleRef", () => {
  const wrapper = shallow(<RealExampleRef />);

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should not show MultipleCustomHooks component", () => {
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  it("should show MultipleCustomHooks component", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
