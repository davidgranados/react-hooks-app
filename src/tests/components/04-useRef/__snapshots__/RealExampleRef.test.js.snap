// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on RealExampleRef should match snapshot 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
